DUnitX.Assert.pas

```diff
@@ -48,21 +49,22 @@ interface
   Assert = class
   private
     class var fOnAssert: TProc;
-    class var fTestFailure: ExceptClass;
```

Comment:
I didn't realize that is why this was done.  There weren't any comments explaining that, so I thought it was just something left over from the past, or a way around a circular reference.  I had to eliminate the "generic" exception because I couldn't get away with just message - I needed to add custom data to the exception.  I could revert the changes and make some new exception classes, but dumb down to the lowest common denominator if any class var is nil (for instance, if fTestFailureStrCompare = nil, raise fTestFailure) and I think that would satisfy the requirement.